blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/EnumValidator.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2015 Rackspace
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.service;
+
+import com.netflix.astyanax.connectionpool.exceptions.ConnectionException;
+import com.netflix.astyanax.model.ColumnList;
+import com.rackspacecloud.blueflood.io.AstyanaxReader;
+import com.rackspacecloud.blueflood.io.AstyanaxWriter;
+import com.rackspacecloud.blueflood.io.DiscoveryIO;
+import com.rackspacecloud.blueflood.io.SearchResult;
+import com.rackspacecloud.blueflood.types.BluefloodEnumRollup;
+import com.rackspacecloud.blueflood.types.IMetric;
+import com.rackspacecloud.blueflood.types.Locator;
+import com.rackspacecloud.blueflood.types.PreaggregatedMetric;
+import com.rackspacecloud.blueflood.utils.ModuleLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.*;
+
+/** This class handles collecting enum values for specific metrics, via their locators, and checking whether
+    a uniqueness count of the enum values have reached a certain threshold.  If it has, the class mark the metric
+    as bad by inserting its locator into the proper cassandra column family.  If it hasn't reached the threshold, then it
+    will create or update the elasticsearch "enums" index for the metric.
+**/
+public class EnumValidator implements Runnable {
+
+    private static final Logger log = LoggerFactory.getLogger(EnumValidator.class);
+    private static final Configuration config = Configuration.getInstance();
+    private static final DiscoveryIO discoveryIO = (DiscoveryIO) ModuleLoader.getInstance(DiscoveryIO.class, CoreConfig.ENUMS_DISCOVERY_MODULES);
+    private static final int ENUM_UNIQUE_VALUES_THRESHOLD = config.getIntegerProperty(CoreConfig.ENUM_UNIQUE_VALUES_THRESHOLD);
+    private Set<Locator> locators;
+
+    public EnumValidator(Set<Locator> locators) {
+        this.locators = locators;
+    }
+
+    @Override
+    public void run() {
+
+        if (locators == null) return;
+        Map<Locator, ColumnList<Long>> cfMetrics = AstyanaxReader.getInstance().getStringAndHashMappingForEnums(new ArrayList(locators));
```

Comment:
@VinnyQ would it help if this method directly returns the HashMap with the mapping of Locator to List<Strings>